reference-implementation/README.md

```diff
@@ -12,30 +12,47 @@ Test coverage is not complete, but we do aim for it to be. Adding tests would be
 
 You can check the test coverage at any time by running `npm run coverage` in this folder.
 
-### Original Tests
+To run all tests (and the lint step), run `npm test` in this folder.
 
-The original tests are written using the [tape](https://github.com/substack/tape) framework.
+### Legacy tests
 
-- To run them, type `npm test` in this folder.
-- To run a specific test file, type e.g. `node test-runner test/readable-stream.js`.
+The original tests are written using the [tape](https://github.com/substack/tape) framework. They live in the `test` subfolder of this folder.
+
+- To run them, type `npm run legacy-test` in this folder.
+- To run a specific test file, type e.g. `node run-tests.js test/readable-stream.js`.
 - To run a specific test, change the source file's `test(...)` call to `test.only(...)`, then run `npm test`.
 
-### Web Platform Tests
+### Web platform tests
+
+After we started implementing this in browsers, we realized that we should probably have written the tests in standard [web platform tests](https://github.com/w3c/web-platform-tests) format. Fixing this has been an ongoing effort, with major help from various community contributors.
+
+- To run the web platform tests (including both the upstream ones and the to-upstream ones), type `npm run wpt` in this folder.
+- To run specific test files, you can use a glob pattern, rooted at the streams directory: `npm run wpt -- "writable-streams/**"`
 
-After we started implementing this in browsers, we realized that we should probably have written the tests in standard [web platform tests](https://github.com/w3c/web-platform-tests) format. Fixing this has been an ongoing effort, with major help from WebKit contributors, who ported the initial set of tests from our original format.
+The test runner here is a Node.js emulated-DOM environment, with the reference implementation loaded into it.
 
-The web platform tests for streams are found in the [streams directory](https://github.com/w3c/web-platform-tests/tree/master/streams) of the web platform tests repository, and maintained via pull requests to that repository.
+#### Upstream web platform tests
 
-For local development while working on the spec, you can run
+The web platform tests for streams are found in the [streams directory](https://github.com/w3c/web-platform-tests/tree/master/streams) of the web platform tests repository, and maintained via pull requests to that repository. They are then pulled into this repository via a [Git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules).
+
+This means that in order to land a test change for these tests, you'll need to make a pull request to the web platform tests repository, and then update the submodule pointer in this repository (probably in the same pull request as your spec change). That can be done via the command
```

Comment:
Yeah, I tried to allude to that a couple paragraphs down, but we'll see how well it works in practice.